e2e/callback_test.go

```diff
@@ -0,0 +1,372 @@
+	require.NoError(t, err)
+	return *resp.Owner
+}
+
+func queryTesterReceivedNftContract(t *testing.T, chain *wasmibctesting.TestChain, tester string) string {
+	resp := ""
+	testerReceivedQuery := TesterReceivedNftContractQuery{
+		GetReceivedNftContract: EmptyData{},
+	}
+	err := chain.SmartQuery(tester, testerReceivedQuery, &resp)
+	require.NoError(t, err)
+	return resp
+}
+
+func queryTesterReceivedErr(t *testing.T, chain *wasmibctesting.TestChain, tester string) error {
+	resp := TesterResponse{}
+	testerReceivedQuery := TesterReceivedQuery{
+		GetReceivedCallback: EmptyData{},
+	}
+	err := chain.SmartQuery(tester, testerReceivedQuery, &resp)
+	return err
+}
+
+func (suite *CbTestSuite) TestSuccessfulTransfer() {
+	memo := callbackMemo(nftSentCb(), "", nftReceivedCb(), "")
+	sendIcsFromChainA(suite, suite.cw721A.String(), "2", memo, true)
+
+	// Query the owner of NFT on cw721
+	chainAOwner := queryGetOwnerOf(suite.T(), suite.chainA, suite.cw721A.String(), "2")
+	require.Equal(suite.T(), chainAOwner, suite.bridgeA.String())
+	chainBOwner := queryGetOwnerOf(suite.T(), suite.chainB, suite.cw721B.String(), "2")
+	require.Equal(suite.T(), chainBOwner, suite.testerB.String())
+
+	// We query the data we have on the tester contract
+	// This ensures that the callbacks are called after all the messages was completed
+	// and the transfer was successful
+	testerDataOwnerA := queryTesterSent(suite.T(), suite.chainA, suite.testerA.String())
+	require.Equal(suite.T(), testerDataOwnerA, suite.bridgeA.String())
+	testerNftContract := queryTesterReceivedNftContract(suite.T(), suite.chainB, suite.testerB.String())
+	require.Equal(suite.T(), testerNftContract, suite.cw721B.String())
+	testerDataOwnerB := queryTesterReceived(suite.T(), suite.chainB, suite.testerB.String())
+	require.Equal(suite.T(), testerDataOwnerB, suite.testerB.String())
+}
+
+func (suite *CbTestSuite) TestSuccessfulTransferWithReceivers() {
+	memo := callbackMemo(nftSentCb(), suite.testerA.String(), nftReceivedCb(), suite.testerB.String())
+
+	// Send NFT to chain B
+	ics721Nft(suite.T(), suite.chainA, suite.path, suite.coordinator, suite.cw721A.String(), "3", suite.bridgeA, suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress(), memo)
+
+	// Query the owner of NFT on cw721
+	chainAOwner := queryGetOwnerOf(suite.T(), suite.chainA, suite.cw721A.String(), "3")
+	require.Equal(suite.T(), chainAOwner, suite.bridgeA.String())
+	chainBOwner := queryGetOwnerOf(suite.T(), suite.chainB, suite.cw721B.String(), "3")
+	require.Equal(suite.T(), chainBOwner, suite.chainB.SenderAccount.GetAddress().String())
+
+	// We query the data we have on the tester contract
+	// This ensures that the callbacks are called after all the messages was completed
+	// and the transfer was successful
+	testerDataOwnerA := queryTesterSent(suite.T(), suite.chainA, suite.testerA.String())
+	require.Equal(suite.T(), testerDataOwnerA, suite.bridgeA.String())
```

Comment:
Good. Same result for this with ack and receive callback, as previous test with just receive callback.